packages/cojson-storage/src/managerAsync.ts

```diff
@@ -157,54 +147,56 @@ export class StorageManagerAsync {
 
         collectNewTxs({
           newTxsInSession,
-          newContentMessages,
+          contentMessage,
           sessionRow,
           firstNewTxIdx: idx,
           signature: signature.signature,
         });
 
         idx = signature.idx + 1;
+
+        if (signatures.length > 1) {
+          await this.sendContentMessage(coValueRow, contentMessage);
+          contentMessage = {
+            action: "content",
+            id: coValueRow.id,
+            header: coValueRow.header,
+            new: {},
+            priority: cojsonInternals.getPriorityFromHeader(coValueRow.header),
+          } satisfies CojsonInternalTypes.NewContentMessage;
+        }
       }
     }
 
-    this.loadedCoValues.add(coValueRow.id);
+    return this.sendContentMessage(coValueRow, contentMessage);
+  }
+
+  async sendContentMessage(
+    coValueRow: StoredCoValueRow,
+    contentMessage: CojsonInternalTypes.NewContentMessage,
+  ) {
+    if (Object.keys(contentMessage.new).length === 0) {
+      return;
+    }
 
     const dependedOnCoValuesList = getDependedOnCoValues({
       coValueRow,
-      newContentMessages,
+      newContentMessages: [contentMessage],
     });
```

Comment:
Going to check the dependencies before sending each content message.

This way we ensure that the order is respected while having the possiblity of doing streaming.